torch/_export/utils.py

```diff
@@ -45,6 +58,106 @@
 }
 
 
+# Our strategy for deciding if we can preserve a op is following:
+# 1. The op should be known statically that it is functional
+# 2. If it is maybe aliasing, we decompose because we must know if an op
+#    is mutating or aliasing.
+# TODO (tmanlaibaatar) make this utility function and share it with functional_tensor
+# decomp part. (https://github.com/pytorch/pytorch/issues/129431)
+def _assert_valid_to_preserve(op_overload):
```

Comment:
so - in the new design, is there still any notion from the user API of "ops to preserve"?

I thought that this PR removes it from the user API (and makes it an implementation detail) - in which case we should be able to remove these user assertions?

Reply:
User can still specify a custom decomp behaivour, so we need to still run these asserts. I will create _check_valid_to_preserve util function for the cases where we just need to check.

Reply:
oh, you're saying we will still / only raise these asserts if the user tries to override one of these decomps with their own? That sounds good